Add migration guide (#897)
flowforge/flowforge: docs/user/migration.md

Presently it is not possible to install older versions of a package into Node-RED so your flows will need to run on the latest version of each package, you may want to consider updating before you migrate.

### Backup your flows

From the Node-RED editor goto the hamburger menu, and select **_Export_**![](./images/mig_export_menu.png)

In the modal that opens ensure that you  have**_ all flows _**selected and then click the download button. This will download a file called **_flows.json_** to your browser, keep it safe.

![](https://lh5.googleusercontent.com/rY4vl2YoHINNMoK0MTTrLITrBJgr9Detxd6w3KRJQWXkvRoEv6625Ope0QG3WmCTtRHeJFOk6KCoBfniONIjSP1lxNTltEROkpys8j0CsHuPqKM74HoNMqt0AYufg8z9JRDsrJOa197op8hXwZzgiDU)

### Credentials.

It is not possible to get credential details from the Node-RED editor once they have been entered

There are 2 options available:

Firstly you can walk through your flows manually  and check any credential fields, then make sure you have access to the information in another source.

> Follow the instructions at <https://www.hardill.me.uk/wordpress/2021/02/17/viewing-node-red-credentials/> if you have access to the command line for the node-red install.

Reviewer: ```suggestion
Follow the instructions at <https://www.hardill.me.uk/wordpress/2021/02/17/viewing-node-red-credentials/> if you have access to the command line for the Node-RED install.
```

### Environment Variables

You can use this flow <https://flows.nodered.org/flow/8ebfe9ae218aa5105e7da13db14ac272> to dump a list of your environment variables into the debug window.

### Static Files

> Check your settings.js file to see if \`httpStatic\` has been set, if so then check for any files in this path.FlowForge does not currently support serving static files so you will need to find alternative hosting for these such as AWS S3.

Reviewer: ```suggestion
Check your `settings.js` file to see if `httpStatic` has been set, if so then check for any files in this path. FlowForge does not currently support serving static files so you will need to find alternative hosting for these such as AWS S3.
```

## Import to FlowForge

> ### Create a new flowforge project 

Reviewer: ```suggestion
### Create a new FlowForge project 
```

> From your team page click create project and assign a suitable name to the project, 

Reviewer: ```suggestion
From your team page click create project and assign a suitable name to the project.
```

### Environment Variables.

> If your project uses environment variables you can configure these in the flowforge application under the Project > Settings >Environment tabs.

Reviewer: ```suggestion
If your project uses environment variables you can configure these in the FlowForge application under the Project > Settings > Environment tabs.
```

### Install Nodes

> Open the editor and goto the manage palette option in the hamburger menu. 

Reviewer: ```suggestion
Open the editor and go to the manage palette option in the hamburger menu. 
```

> Now install each of the nodes that you identified in the export above

Reviewer: ```suggestion
Now install each of the nodes that you identified in the export earlier.
```

### Import Flows

> From the hamburger menu select Import

Reviewer: ```suggestion
From the hamburger menu select "Import". Either paste your `flows.json` or upload the file. Click "Import".
Check that you don’t get any errors about missing nodes. If you do, go back and check that you have installed all the required nodes.

When there are no errors click "Deploy" to save the flows. You may want to ensure your flows are stopped to the "Deploy" action does not start your flows yet.
```
